spec/requests/trains_request_spec.rb

```diff
@@ -29,4 +49,92 @@
       expect(JSON.parse(response.body)).to eq(expected_response)
     end
   end
+
+  describe 'POST /v1/trains' do
+    context 'with valid attributes' do
+      it 'returns http status 201' do
+        post v1_trains_path params: { train: expected_response }
+        expect(response).to be_created
+        expect(JSON.parse(response.body)).to eq({"message"=>"#{train.name} successfully created"})
+      end
+    end
+
+    context 'with invalid attributes' do
+      it 'returns http status 422' do
+        post v1_trains_path params: { train: invalid_attributes }
+        expect(response).to have_http_status(422)
+        expect(JSON.parse(response.body)).to eq({
+          "errors"=>[
+            {
+              "name"=>[
+                "can't be blank"
+              ],
+              "train_model_name"=>[
+                "can't be blank"
+              ],
+              "number_of_cars"=>[
+                "can't be blank",
+                "is not a number"
+              ],
+              "max_weight_capacity"=>[
+                "can't be blank",
+                "is not a number"
+              ],
+              "active"=>[
+                "is not included in the list"
+              ]
+            }
+          ]
+        })
+      end
+    end
+  end
+
+  describe 'PUT /v1/trains/1' do
+    context 'with valid attributes' do
+      it 'returns http status 200' do
+        put v1_train_path(train), params: { train: valid_attributes }
+        expect(response).to be_ok
+        expect(JSON.parse(response.body)).to eq({"message"=>"name successfully updated"})
```

Comment:
assert the train in question was actually updated too please.
we are just asserting that the messages appear as expected, but not that we have update the train record we pass in